app/controllers/organisations_controller.rb

```diff
@@ -94,10 +94,42 @@ def edit
     end
   end
 
+  def deactivate
+    authorize @organisation
+
+    render "toggle_active", locals: { action: "deactivate" }
+  end
+
+  def reactivate
+    authorize @organisation
+
+    render "toggle_active", locals: { action: "reactivate" }
+  end
+
   def update
-    if current_user.data_coordinator? || current_user.support?
+    if (current_user.data_coordinator? && org_params[:active].nil?) || current_user.support?
       if @organisation.update(org_params)
-        flash[:notice] = I18n.t("organisation.updated")
+        case org_params[:active]
+        when "false"
+          @organisation.users.filter_by_active
+                       .update_all(
+                         active: false,
+                         confirmed_at: nil,
+                         sign_in_count: 0,
+                         initial_confirmation_sent: false,
+                         reactivate_with_organisation: true,
+                       )
```

Comment:
I assume we set all of these when we deactivate user independently of org? I like `reactivate_with_organisation`!